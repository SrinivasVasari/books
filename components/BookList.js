import React, { useContext, useState, Fragment } from "react";
import { useBooksContext } from "../context/context";
import BookDetails from "./BookDetails";

const BookList = () => {
  const { state, dispatch } = useBooksContext();
  const [selected, setSelected] = useState(null);
  const selectedBook = state && state.books.filter((book) => {
    if (book.id === selected) {
      return book;
    }
  });

  return (
    <Fragment>
      <ul className="books-list">
        {state && state.books.map((book) => {
          const { bookName, bookAuthor } = book.books;
          return (
            <li className="book" key={book.id} onClick={(e) => setSelected(book.id)}>
              <div className="header">
                <h5>{bookName}</h5>
                <span className="author"><i>__{bookAuthor}</i></span>
              </div>
              <div className="btn-container">
                <a
                  onClick={() =>
                    dispatch({ type: "SET_CURRENT_BOOK", payload: book })
                  }
                  className="btn-floating btn-small waves-effect waves-light blue"
                >
                    <i className="material-icons">edit</i>
                </a>
                <a
                  onClick={() =>
                    dispatch({ type: "DELETE_BOOK", payload: book.id })
                  }
                  className="btn-floating btn-small waves-effect waves-light orange"
                >
                  <i className="material-icons">delete</i>
                </a>
              </div>
            </li>
          );
        })}
      </ul>

      <div className="bookDetails">
        <BookDetails book={state && selectedBook[0]} bookId={selected} />
      </div>
    </Fragment>
  );
};

export default BookList;
