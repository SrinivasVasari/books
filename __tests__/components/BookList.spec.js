import React from "react";
import { shallow, mount } from "enzyme";
import toJson from 'enzyme-to-json';

import BookList from "../../components/BookList";

describe("Edit Book Container", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<BookList />);
  });

  it("should render correctly", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it("initially displays 3 items, first of them done", () => {
    const wrapper = mount(<BookList />);
    expect(wrapper.find("li")).toHaveLength(3);
  });
});
