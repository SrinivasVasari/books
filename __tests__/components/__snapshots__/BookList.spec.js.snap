// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit Book Container should render correctly 1`] = `undefined`;
